cocotb/scheduler.py

```diff
@@ -573,11 +571,10 @@ def queue_function(self, coro):
         # each entry always has a unique thread.
         t, = matching_threads
 
-        @cocotb.coroutine
-        def wrapper():
+        async def wrapper():
             # This function runs in the scheduler thread
             try:
-                _outcome = outcomes.Value((yield coro))
+                _outcome = outcomes.Value((await coro))
```

Comment:
```suggestion
                _outcome = outcomes.Value(await coro)
```
Await has a more convenient place in the grammar than yield, which means the parens aren't requires.